src/N98/Magento/Command/Admin/User/LockCommand.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace N98\Magento\Command\Admin\User;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputArgument;
+
+class LockCommand extends AbstractAdminUserCommand
+{
+    /**
+     * The number of days to lock for (by default)
+     * @var int
+     */
+    const LIFETIME_DEFAULT = 31;
+
+    /**
+     * Setup
+     *
+     * @return void
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('admin:user:lock')
+            ->addArgument('username', InputArgument::REQUIRED, 'Admin username to lock')
+            ->addArgument('lifetime', InputArgument::OPTIONAL, 'Optional - lock lifetime (default one month)')
+            ->setDescription(
+                <<<HELP
+Enforce a lock on an admin user account. Specify the username and an optional lifetime parameter in seconds.
+HELP
+            );
+    }
+
+    /**
+     * @return bool
+     */
+    public function isEnabled()
+    {
+        return $this->getApplication()->isMagentoEnterprise();
+    }
+
+    /**
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     *
+     * @return void
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->detectMagento($output, true);
+        if (!$this->initMagento()) {
+            return;
+        }
+
+        $username = $input->getArgument('username');
+        $lifetime = $input->getArgument('lifetime') ?: $this->daysToSeconds(self::LIFETIME_DEFAULT);
+
+        $user = \Mage::getModel('admin/user')->loadByUsername($username);
+        if (!$user || !$user->getId()) {
+            $output->writeln("<error>Couldn't find admin username '{$username}'</error>");
+            return;
+        }
+
+        \Mage::getResourceModel('enterprise_pci/admin_user')->lock($user->getId(), 0, $lifetime);
```

Comment:
Hey Tom- I had assumed that the enterprise check in isEnabled() would run
automatically and exclude this class from being used. If that's not the
case I'm happy to update!

On Tuesday, 12 April 2016, Tom Klingenberg <notifications@github.com> wrote:

> In src/N98/Magento/Command/Admin/User/LockCommand.php
> <https://github.com/netz98/n98-magerun/pull/810#discussion_r59322010>:
>
> > +    {
> > +        $this->detectMagento($output, true);
> > +        if (!$this->initMagento()) {
> > +            return;
> > +        }
> > +
> > +        $username = $input->getArgument('username');
> > +        $lifetime = $input->getArgument('lifetime') ?: $this->daysToSeconds(self::LIFETIME_DEFAULT);
> > +
> > +        $user = \Mage::getModel('admin/user')->loadByUsername($username);
> > +        if (!$user || !$user->getId()) {
> > +            $output->writeln("<error>Couldn't find admin username '{$username}'</error>");
> > +            return;
> > +        }
> > +
> > +        \Mage::getResourceModel('enterprise_pci/admin_user')->lock($user->getId(), 0, $lifetime);
>
> Just by the name of the model, could it be that enterprise_pci/admin_user
> is not available in community edition? If so, this line would crash (method
> call on a non-object).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/netz98/n98-magerun/pull/810/files/f2a199f512b88c75d9ef699de6f778c3100b7f7f#r59322010>
>


-- 
*Robbie Averill*
LA PETITE MANOUCHE - NZ Gypsy Jazz Guitar Duo
Phone: +64 27 321 4758
contact@lapetitemanouche.com
www.lapetitemanouche.com